lahja/common.py

```diff
@@ -0,0 +1,111 @@
+from abc import (
+    ABC,
+    abstractmethod,
+)
+from typing import (
+    TYPE_CHECKING,
+    Generic,
+    NamedTuple,
+    Optional,
+    Set,
+    Type,
+    TypeVar,
+    Union,
+)
+
+from lahja.misc import (
+    BroadcastConfig,
+)
+
+if TYPE_CHECKING:
+    from lahja.base import EndpointAPI
+
+
+class BaseEvent:
+    __slots__ = ("_origin", "_id")
+
+    @property
+    def is_bound(self) -> bool:
+        return hasattr(self, '_origin')
+
+    def bind(self,
+             endpoint: 'EndpointAPI',
+             id: Optional[str] = None) -> None:
+        """
+        Bind an event to an end
+        """
+        self._origin = endpoint.name
+        self._id = id
+
+    def response_config(self, internal: bool = False) -> BroadcastConfig:
+        if internal:
+            return BroadcastConfig(
+                internal=True,
+                filter_event_id=self._id
+            )
+        else:
+            return BroadcastConfig(
+                filter_endpoint=self._origin,
+                filter_event_id=self._id
+            )
+
+
+TResponse = TypeVar('TResponse', bound=BaseEvent)
+
+
+class BaseRequestResponseEvent(ABC, BaseEvent, Generic[TResponse]):
+    __slots__ = ()
+
+    @staticmethod
+    @abstractmethod
+    def expected_response_type() -> Type[TResponse]:
```

Comment:
@cburgdorf remind me why this was made a `staticmethod` rather than just an attribute/property

Reply:
Good question. I don't see it being accessed statically anywhere. I don't see a reason (maybe there was in the past?) why that shouldn't just become an abstract property